modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/ArtifactClassLoader.java

```diff
@@ -99,4 +100,10 @@ public interface ArtifactClassLoader extends DisposableClassLoader, LocalResourc
    */
   void addShutdownListener(ShutdownListener listener);
 
+  default void setModuleLayerInformationSupplier(Supplier<String> moduleLayerInformationSupplier) {}
+
+  default String getModuleLayerInformation() {
```

Comment:
maybe using an object instead of a String in case we want to add more information in the future?